### DISCUSSION TITLE
Zettel preview: adding graph metadata

### DISCUSSION THREAD

**Discussion by GenrihGrigoryan** [srid/neuron#491] [Ideas]
It would be nice, to have possibility to choose the default preview of generated Neuron site, and to customise it for each page, if needed.
It can be done by adding, properties in frontmatter. Fo example: 'preview-image: [image]' - for image, 'preview-description: [text]' - for for description.

Main goal - to achive preview of page in social media (facebook, telegram, twitter, etc) from this:
<img width="232" alt="Снимок экрана 2020-12-12 в 12 01 14" src="https://user-images.githubusercontent.com/7505178/101979938-757d0d00-3c72-11eb-8ebf-bce7b656d88d.png">

To something like this:
<img width="368" alt="Снимок экрана 2020-12-12 в 12 01 21" src="https://user-images.githubusercontent.com/7505178/101979942-8463bf80-3c72-11eb-9fd2-1b734a66c971.png">

Possible solution - Open Graph Protocol: https://ogp.me

**Comment by GenrihGrigoryan** [comment]
update: 

It can be done, by adding these lines of code in header.html:

```
<meta property="og:image" content="static/black.png">
<meta property="twitter:image" content="static/black.png">
```

Then, the better idea: is possibility to automatically generate image preview with title of black, or white background.
For example, i'm choosing preview background in "neuron.dhall": 
```
use preview-image = True,
preview-background = black
```

And after sharing a link, people will see something like this:
![Rectangle](https://user-images.githubusercontent.com/7505178/101984174-c4399f80-3c90-11eb-967c-317fa8deb8d6.png)

**Comment by srid** [comment]
Neuron already uses Open Graph Protocol. It will use the first paragraph as description, and the first image (if any) as the preview image.  However we can make them overridable in the YAML front matter. It would be a good first contribution.

EDIT: Yea, sitewide OGP configuration can be specified in neuron.dhall. RFC and PR welcome.
